### PR TITLE
Feature: Add unit test workflow

### PR DESCRIPTION
**Pull request by kjohnson**
## Description

This PR implements a workflow using the GitHub Actions pipeline to run add-on unit tests with GiveWP as a dependency.

## Pre-review Checklist

-   [x] Acceptance criteria satisfied and marked in related issue
-   [ ] Relevant `@unreleased` tags included in DocBlocks
-   [x] Includes unit and/or end-to-end tests
-   [x] [Self Review](https://give.gitbook.io/development-manual/devops/github/code-reviews#self-review) of code and UX completed

### CODE REVIEW

File: tests/unit/bootstrap.php [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+	}
+
+	/**
+	 * Uninstall plugin
+	 */
+	public function uninstall_plugin() {
+		give_update_option( 'uninstall_on_delete', 'enabled' );
+
+		// WP_UNINSTALL_PLUGIN is already define while uninstalling give.
+//		include( $this->plugin_dir . '/uninstall.php' );
```
Comment (JasonTheAdams) [inline tests/unit/bootstrap.php]: So do the commented out sections need to be un-commented in order to work? Why not just un-comment them now?

**Comment by JasonTheAdams** [comment]
@kjohnson We may consider renaming the `.github` directory so it's not actually picked up by Github, causing failing tests 👇. The build script can rename it to `.github`. What do you think?

File: tests/unit/bootstrap.php (hunk 1 above)
Comment (kjohnson) [reply tests/unit/bootstrap.php]: My work in GiveWP core suggests that we can now just remove that uninstall hook all together. I'll go ahead and do that now.

Comment (JasonTheAdams) [reply tests/unit/bootstrap.php]: Cool! Same question for the other commented out sections.

Comment (kjohnson) [reply tests/unit/bootstrap.php]: @JasonTheAdams see https://github.com/impress-org/givewp-addon-boilerplate/pull/57/commits/34d89de86d5537987205dec5c9b2e6879e0df7aa

Comment (kjohnson) [reply tests/unit/bootstrap.php]: As for the other commented out sections, those are placeholders showing where to load custom helpers classes.

**Comment by kjohnson** [comment]
@JasonTheAdams good idea! I updated the build to rename to `.github`, see https://github.com/impress-org/givewp-addon-boilerplate/pull/57/commits/5f41e7dcaa860657a1408dad9a43923fc828848f.

File: tests/unit/bootstrap.php [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+
+	/**
+	 * Install recurring.
+	 */
+	public function install_plugin() {
+
+		echo 'Installing ADDON_DOMAIN...' . PHP_EOL;
+
+		// give_{addon}_install();
+	}
```
Comment (JasonTheAdams) [inline tests/unit/bootstrap.php:86]: I'm confused by this. Is this installing other addons (such as recurring)? Or is this required to install this addon? In which case, why is it commented out?

File: tests/unit/bootstrap.php [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+	public function includes() {
+		Give_Unit_Tests_Bootstrap::instance()->includes();
+
+		require_once( $this->tests_dir . '/helpers.php' );
+
+		// test cases
+		// require_once( $this->tests_dir . '/framework/class-{testcase}.php' );
+
+		//Helpers
+		// require_once( $this->tests_dir . '/framework/helpers/class-{helper}.php' );
```
Comment (JasonTheAdams) [inline tests/unit/bootstrap.php:102]: I vote for just leaving these as uncommented so folks can just update the existing files. They can always come in here and remove them.

**Review by JasonTheAdams — CHANGES_REQUESTED** [review]
Probably the biggest thing I'd like to have moving forward is PSR-4 autoloading with our tests so tests can run without having to have includes to make them available. It also reduces the memory of the tests as test numbers increase and only individual tests are run. Add it as an entirely new PSR-4 line to composer. We can use `GiveWPTests\\ADDON_NAME` as the namespace.

File: tests/unit/bootstrap.php (hunk 3 above)
Comment (kjohnson) [reply tests/unit/bootstrap.php:102]: Yeah, I would leave these here as instructions, which can easily be removed if unnecessary.

File: tests/unit/bootstrap.php (hunk 2 above)
Comment (kjohnson) [reply tests/unit/bootstrap.php:86]: Like the test case and helper includes, this is a placeholder for where to add any additional install steps. GiveWP core has `give()->install()`, but it is probably less common for add-ons, which can just hook into GiveWP `before_give_init`, etc.

**Comment by kjohnson** [comment]
Closing this as stale.

**Pull request closed without merging**